sharedbuffers/mapped_struct.py

```diff
@@ -1718,6 +1741,519 @@ def __str__(self):
+    def _add_to(self, out):
+        if self.objlist is None:
+            for i in xrange(64):
+                if self.bitrep_lo & (cython.cast(cython.ulonglong, 1) << i):
+                    out.add(i)
+                elif not self.bitrep_lo >> i:
+                    break
+            for i in xrange(64):
+                if self.bitrep_hi & (cython.cast(cython.ulonglong, 1) << i):
+                    out.add(i + 64)
+                elif not self.bitrep_hi >> i:
+                    break
+            return out
+
+        dcode, objlen, itemsize, offset, _struct = self.objlist._metadata()
+        for i in xrange(len(self)):
+            out.add(self.objlist._c_getitem(i, dcode, objlen, itemsize, offset, _struct, None))
+        return out
+
+    @cython.ccall
+    @cython.locals(pfset='proxied_frozenset')
+    def _union_2(self, seq):
+        if not seq:
+            return self
+        elif self.objlist is None and type(seq) is proxied_frozenset:
+            pfset = cython.cast(proxied_frozenset, seq)
+            if pfset.objlist is None:
+                return proxied_frozenset(
+                    None, self.bitrep_lo | pfset.bitrep_lo, self.bitrep_hi | pfset.bitrep_hi)
+
+        rv = self._add_to(set())
+        rv.update(seq)
+        return frozenset(rv)
+
+    @cython.ccall
+    @cython.locals(pfset='proxied_frozenset')
+    def _intersect_2(self, seq):
+        if self.objlist is None and type(seq) is proxied_frozenset:
+            pfset = cython.cast(proxied_frozenset, seq)
+            if pfset.objlist is None:
+                return proxied_frozenset(
+                    None, self.bitrep_lo & pfset.bitrep_lo, self.bitrep_hi & pfset.bitrep_hi)
+        elif not seq:
+            return proxied_frozenset(None, 0, 0)
+
+        rv = self._add_to(set())
+        rv.intersection_update(seq)
+        return frozenset(rv)
+
+    @cython.ccall
+    @cython.locals(pfset='proxied_frozenset')
+    def _diff_2(self, seq):
+        if not seq:
+            return self
+        elif self.objlist is None and type(seq) is proxied_frozenset:
+            pfset = cython.cast(proxied_frozenset, seq)
+            if pfset.objlist is None:
+                return proxied_frozenset(
+                    None, self.bitrep_lo & ~pfset.bitrep_lo, self.bitrep_hi & ~pfset.bitrep_hi)
+
+        rv = self._add_to(set())
+        rv.difference_update(seq)
+        return frozenset(rv)
+
+    @cython.ccall
+    @cython.locals(pfset='proxied_frozenset')
+    def _symdiff_2(self, seq):
+        if not seq:
+            return self
+        elif self.objlist is None and type(seq) is proxied_frozenset:
+            pfset = cython.cast(proxied_frozenset, seq)
+            if pfset.objlist is None:
+                return proxied_frozenset(
+                    None, self.bitrep_lo ^ pfset.bitrep_lo, self.bitrep_hi ^ pfset.bitrep_hi)
+
+        rv = self._add_to(set())
+        rv.symmetric_difference_update(seq)
+        return frozenset(rv)
```

Comment:
I don't particularly care about this one, but you can do it as: `(self - seq) | (seq - self)` I think